C/c4Database.cc

```diff
@@ -115,11 +117,11 @@ C4Database* c4db_openNamed(C4String name,
                            const C4DatabaseConfig2 *config,
                            C4Error *outError) C4API
 {
-    if (!ensureConfigDirExists(config, outError))
-        return nullptr;
-    FilePath path = dbPath(name, config->parentDirectory);
-    C4DatabaseConfig oldConfig = newToOldConfig(config);
-    return tryCatch<C4Database*>(outError, [=] {
+    return tryCatch<C4Database*>(outError, [=]() -> C4Database* {
+        if (!ensureConfigDirExists(config, outError))
+            return nullptr;
```

Comment:
Why was this moved inside the try catch if in the same PR you also added "noexcept" to the signature of ensureConfigDirExists?

Reply:
The important fix is that the call to `dbPath` is now inside the try/catch block.

I added `noexcept` to ensureConfigDirExists just to clarify that it can't throw an exception (because it's surrounded in a try/catch block.)

Reply:
Oh I didn't notice that it just wasn't there at all before as opposed to the other block where try catch was modified (copyNamed).  Nevermind.